src/test/java/com/ericsson/gerrit/plugins/eiffel/messaging/RetryRequestTest.java

```diff
@@ -0,0 +1,125 @@
+import org.powermock.reflect.Whitebox;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.retry.RetryCallback;
+import org.springframework.retry.RetryContext;
+import org.springframework.retry.support.RetryTemplate;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import com.ericsson.eiffelcommons.utils.HttpRequest;
+import com.ericsson.eiffelcommons.utils.ResponseEntity;
+import com.ericsson.gerrit.plugins.eiffel.configuration.EiffelPluginConfiguration;
+import com.ericsson.gerrit.plugins.eiffel.configuration.RetryConfiguration;
+import com.ericsson.gerrit.plugins.eiffel.events.EiffelSourceChangeCreatedEvent;
+import com.ericsson.gerrit.plugins.eiffel.events.EiffelSourceChangeSubmittedEvent;
+import com.ericsson.gerrit.plugins.eiffel.events.generators.EiffelSourceChangeSubmittedEventGenerator;
+import com.ericsson.gerrit.plugins.eiffel.exceptions.HttpRequestFailedException;
+import com.ericsson.gerrit.plugins.eiffel.listeners.ChangeMergedEventListener;
+import com.google.gerrit.server.events.ChangeMergedEvent;
+
+@RunWith(PowerMockRunner.class)
+@PowerMockRunnerDelegate(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = RetryConfiguration.class)
+@PrepareForTest({ EiffelSourceChangeSubmittedEventGenerator.class,
+        ChangeMergedEventListener.class })
+public class RetryRequestTest {
+    private EiffelPluginConfiguration pluginConfig;
+    private HttpRequest httpRequest;
+    private ResponseEntity response;
+    private ChangeMergedEvent changeMergedEvent;
+    private MutableInt counter = new MutableInt();
+    private EiffelSourceChangeSubmittedEvent eiffelEvent = new EiffelSourceChangeSubmittedEvent();
+
+    private static final String EIFFEL_TYPE = "EiffelSourceChangeCreatedEvent";
+    private static final int STATUS_ERROR = HttpStatus.SC_INTERNAL_SERVER_ERROR;
+    private static final String PLUGIN_NAME = "Eiffel-Integration";
+    private static final File FILE_DIR = new File("");
+
+    @Autowired
+    private RetryTemplate retryTemplate;
+
+    @Before
+    public void beforeTest() throws IOException, URISyntaxException {
+        setUpMockObjects();
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void testRetryLogic() throws URISyntaxException, IOException {
+        setUpMockActions();
+        counter.setValue(0);
+
+        EiffelEventSender sender = new EiffelEventSender(pluginConfig, httpRequest);
+        sender.setEiffelEventMessage(new EiffelSourceChangeCreatedEvent());
+        sender.setEiffelEventType(EIFFEL_TYPE);
+        retryTemplate.execute(new RetryCallback<Void, HttpRequestFailedException>() {
+            @Override
+            public Void doWithRetry(RetryContext context) {
+                counter.setValue(counter.getValue().intValue() + 1);
+                sender.send();
+                return null;
+            }
```

Comment:
Good that you test the logic but you are not testing the cod in `PatchsetCreatedEventListener` and `ChangeMergedEventListener`

Reply:
testPrepareAndSendEiffelEvent tests the ChangeMergedEventListener. I don't do the same for PatchsetCreatedEventListener though.

Reply:
If you move the duplicated code you could only have one place to test

Reply:
I test the send method in AbstractEventListener now.